### PR TITLE
Tasks 492 493 495 496 497

### PR DESCRIPTION
**Pull request by ThanhNIT**

### CODE REVIEW

File: src/main/java/g0401_0500/s0492_construct_the_rectangle/readme.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+Return _an array `[L, W]` where `L` and `W` are the length and width of the web page you designed in sequence._
+
+**Example 1:**
+
+**Input:** area = 4
+
+**Output:** [2,2]
+
+**Explanation:** The target area is 4, and all the possible ways to construct it are [1,4], [2,2], [4,1]. But according to requirement 2, [1,4] is illegal; according to requirement 3, [4,1] is not optimal compared to [2,2]. So the length L is 2, and the width W is 2.
```
Comment (javadev) [inline src/main/java/g0401_0500/s0492_construct_the_rectangle/readme.md:19]: It looks like a multi line block.

File: src/main/java/g0401_0500/s0495_teemo_attacking/readme.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+Return _the **total** number of seconds that Ashe is poisoned_.
+
+**Example 1:**
+
+**Input:** timeSeries = [1,4], duration = 2
+
+**Output:** 4
+
+**Explanation:** Teemo's attacks on Ashe go as follows: - At second 1, Teemo attacks, and Ashe is poisoned for seconds 1 and 2. - At second 4, Teemo attacks, and Ashe is poisoned for seconds 4 and 5. Ashe is poisoned for seconds 1, 2, 4, and 5, which is 4 seconds in total.
```
Comment (javadev) [inline src/main/java/g0401_0500/s0495_teemo_attacking/readme.md:17]: It looks like a multi line block.

File: src/main/java/g0401_0500/s0496_next_greater_element_i/readme.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+Return _an array_ `ans` _of length_ `nums1.length` _such that_ `ans[i]` _is the **next greater element** as described above._
+
+**Example 1:**
+
+**Input:** nums1 = [4,1,2], nums2 = [1,3,4,2]
+
+**Output:** [-1,3,-1]
+
+**Explanation:** The next greater element for each value of nums1 is as follows: - 4 is underlined in nums2 = [1,3,4,2]. There is no next greater element, so the answer is -1. - 1 is underlined in nums2 = [1,3,4,2]. The next greater element is 3. - 2 is underlined in nums2 = [1,3,4,2]. There is no next greater element, so the answer is -1.
```
Comment (javadev) [inline src/main/java/g0401_0500/s0496_next_greater_element_i/readme.md]: It looks like a multi line block.

File: src/main/java/g0401_0500/s0497_random_point_in_non_overlapping_rectangles/readme.md [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+**Example 1:**
+
+![](https://assets.leetcode.com/uploads/2021/07/24/lc-pickrandomrec.jpg)
+
+**Input** ["Solution", "pick", "pick", "pick", "pick", "pick"] [[[[-2, -2, 1, 1], [2, 2, 4, 6]]], [], [], [], [], []]
+
+**Output:** [null, [1, -2], [1, -1], [-1, -2], [-2, -2], [0, 0]]
+
+**Explanation:** Solution solution = new Solution([[-2, -2, 1, 1], [2, 2, 4, 6]]); solution.pick(); // return [1, -2] solution.pick(); // return [1, -1] solution.pick(); // return [-1, -2] solution.pick(); // return [-2, -2] solution.pick(); // return [0, 0]
```
Comment (javadev) [inline src/main/java/g0401_0500/s0497_random_point_in_non_overlapping_rectangles/readme.md]: It looks like a multi line block.

File: src/main/java/g0401_0500/s0492_construct_the_rectangle/readme.md (hunk 1 above)
Comment (ThanhNIT) [reply src/main/java/g0401_0500/s0492_construct_the_rectangle/readme.md:19]: It is one block

File: src/main/java/g0401_0500/s0493_reverse_pairs/Solution.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        return mergeSort(nums, 0, nums.length - 1);
+    }
+
+    private int mergeSort(int[] nums, int start, int end) {
+        if (start >= end) {
+            return 0;
+        }
+        int mid = start + (end - start) / 2;
+        int cnt = mergeSort(nums, start, mid) + mergeSort(nums, mid + 1, end);
+        for (int i = start, j = mid + 1; i <= mid; i++) {
```
Comment (javadev) [inline src/main/java/g0401_0500/s0493_reverse_pairs/Solution.java]: Please move definition for j outside `for` block.

File: src/main/java/g0401_0500/s0493_reverse_pairs/Solution.java [hunk 6]
```diff
@@ -18,9 +18,10 @@ private int mergeSort(int[] nums, int start, int end) {
         }
         int mid = start + (end - start) / 2;
         int cnt = mergeSort(nums, start, mid) + mergeSort(nums, mid + 1, end);
-        for (int i = start, j = mid + 1; i <= mid; i++) {
+        for (int i = start; i <= mid; i++) {
             // it has to be 2.0 instead of 2, otherwise it's going to stack overflow, i.e. test3 is
             // going to fail
+            int j = mid + 1;
```
Comment (javadev) [inline src/main/java/g0401_0500/s0493_reverse_pairs/Solution.java:24]: It should be defined before `for` block.

**Comment by javadev** [comment]
👍

**Pull request merged**